test/test_valgrind.sh

```diff
@@ -5,27 +5,35 @@
 
 set -e
 
-TOOL=$1
+WORKSPACE=$1
+BUILD_DIR=$2
+TOOL=$3
 
 function print_usage() {
-	echo "$(basename $0) - run UMF tests under valgrind tools (memcheck, drd or helgrind)"
-	echo "This script must be run in the UMF build directory. It looks for './test/umf_test-*' test executables."
-	echo "Usage: $(basename $0) <memcheck|drd|helgrind>"
+	echo "$(basename $0) - run all UMF tests under a valgrind tool (memcheck, drd or helgrind)"
+	echo "This script looks for './test/umf_test-*' test executables in the UMF build directory."
+	echo "Usage: $(basename $0) <workspace_dir> <build_dir> <memcheck|drd|helgrind>"
 }
 
 if ! valgrind --version > /dev/null; then
 	echo "error: valgrind not found"
 	exit 1
 fi
 
-if [ $(ls -1 ./test/umf_test-* 2>/dev/null | wc -l) -eq 0 ]; then
-	echo "error: UMF tests ./test/umf_test-* not found (perhaps wrong directory)"
+if [ "$3" == "" ]; then
+	echo -e "error: too few arguments\n"
 	print_usage
 	exit 1
 fi
 
-if [ "$TOOL" == "" ]; then
-	echo "error: valgrind tool is missing"
+if [ ! -f $WORKSPACE/README.md ]; then
+	echo -e "error: incorrect <workspace_dir>: $WORKSPACE\n"
+	print_usage
+	exit 1
+fi
+
+if [ $(ls -1 ${BUILD_DIR}/test/umf_test-* 2>/dev/null | wc -l) -eq 0 ]; then
+	echo -e "error: UMF tests ${BUILD_DIR}/test/umf_test-* not found (maybe incorrect build directory: ${BUILD_DIR})\n"
```

Comment:
you could perhaps suggest instead "(maybe tests not built)" 😉

Reply:
Done